compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -2134,32 +2144,24 @@ where
         TyAndLayout { ty: this.ty, layout }
     }
 
-    fn field(this: TyAndLayout<'tcx>, cx: &C, i: usize) -> C::TyAndLayout {
-        enum TyMaybeWithLayout<C: LayoutOf> {
-            Ty(C::Ty),
-            TyAndLayout(C::TyAndLayout),
+    fn ty_and_layout_field(this: TyAndLayout<'tcx>, cx: &C, i: usize) -> TyAndLayout<'tcx> {
```

Comment:
Hm, would there be value in making sure the naming between this and the `ty_and_layout_FOR_variant` is consistent?

Reply:
The prefix `ty_and_layout_` is consistent between the two, IIRC. If you mean `TyAndLayout::for_variant` should be renamed, I guess `as_variant` or `variant_downcast` might be okay, but I don't have a strong preference.

Reply:
Yeah I mean `ty_and_layout_for_variant`+`ty_and_layout_for_field` or just `ty_and_layout_variant`+`ty_and_layout_field`.